### PR TITLE
[kn-admin]Add support to enable or disable scale-to-zero

### PR DESCRIPTION
**Pull request by zhanggbj**
**User Story**: As a Knative administrator, I want to enable and disable scale to zero for Knative platform.

It might be desirable for Knative admins to enable or disable **scale-to-zero** for a specific Knative installation frequently, which is also discussed in https://github.com/knative/client-contrib/issues/18. The kn-admin plugin will enable/disable this feature via the CLI instead of modifying ConfigMap. We can use this PR for further discussion on it.  Thanks!

Note: please merge PR #25 first and then I'll rebase this one on it.

PR: 
- Init autoscaling subcommand
- support scale-to-zero config

### CODE REVIEW

File: plugins/admin/core/root.go [hunk 1]
```diff
@@ -25,8 +25,9 @@ import (
 	"k8s.io/client-go/tools/clientcmd"
 	"knative.dev/client-contrib/plugins/admin/pkg"
 	"knative.dev/client-contrib/plugins/admin/pkg/command"
+	autoscaling "knative.dev/client-contrib/plugins/admin/pkg/command/autoscaling"
 	"knative.dev/client-contrib/plugins/admin/pkg/command/domain"
-	private_registry "knative.dev/client-contrib/plugins/admin/pkg/command/registry"
+	registry "knative.dev/client-contrib/plugins/admin/pkg/command/registry"
```
Comment (maximilien) [inline plugins/admin/core/root.go:30]: Why are you combining changes to `registry` with this PR which is titled `... enable or disable scale-to-zero`?

Unless these "registry" changes are needed for scale-to-zero I would move them to a different PR.

File: plugins/admin/pkg/command/autoscaling/autoscaling.go [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+)
+
+// domainCmd represents the domain command
+func NewAutoscalingCmd(p *pkg.AdminParams) *cobra.Command {
+	var AutoscalingCmd = &cobra.Command{
+		Use:   "autoscaling",
+		Short: "Manage autoscaling config",
+		Long: `Manage autoscaling provided by Knative Pod Autoscaler (KPA). For example:
+
+kn admin autoscaling set - to manage autoscaling config`,
```
Comment (maximilien) [inline plugins/admin/pkg/command/autoscaling/autoscaling.go:29]: What about also adding the `unset` example too?

File: plugins/admin/pkg/command/autoscaling/set.go [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+		Use:   "set",
+		Short: "set autoscaling config",
+		Long: `set autoscaling config provided by Knative Pod Autoscaler (KPA)
+
+For example:
+# To enable or disable scale-to-zero
+kn admin autoscaling set --scale-to-zero true
+`,
+		Run: func(cmd *cobra.Command, args []string) {
+			currentCm := &corev1.ConfigMap{}
```
Comment (maximilien) [inline plugins/admin/pkg/command/autoscaling/set.go:48]: name this `currentCmd` not `currentCm` which could be confusing later

File: plugins/admin/core/root.go (hunk 1 above)
Comment (zhanggbj) [reply plugins/admin/core/root.go:30]: @maximilien  This change is not required here. Will update it in another PR related to registry. Thanks!

File: plugins/admin/pkg/command/autoscaling/autoscaling.go (hunk 2 above)
Comment (zhanggbj) [reply plugins/admin/pkg/command/autoscaling/autoscaling.go:29]: Sure, need a refine of how to enable/disable scale-to zero and will also add examples here, too

File: plugins/admin/pkg/command/autoscaling/set.go (hunk 3 above)
Comment (zhanggbj) [reply plugins/admin/pkg/command/autoscaling/set.go:48]: as it is short for current ConfigMap, so I think we can keep it as `currentCm`

**Comment by zhanggbj** [comment]
Thanks for your code review!

@rhuss Thanks for point it out. I'll refine this PR to follow the same UX with cli to use `--scale-to-zero` and `--no-scale-to-zero`.

@maximilien Same with above will refine it^^ for `unset`. And I'm working on another item to fill up UT and e2e test #14. So for the future PR, will raise them with test files.

**Comment by zhanggbj** [comment]
All comments addressed, it's ready for review, thanks!

File: plugins/admin/go.mod [hunk 4]
```diff
@@ -7,7 +7,8 @@ require (
 	github.com/spf13/cobra v0.0.5
 	github.com/spf13/viper v1.6.2
 	gotest.tools v2.2.0+incompatible
-	k8s.io/api v0.17.3
-	k8s.io/apimachinery v0.17.3
-	k8s.io/client-go v0.17.0
+	k8s.io/api v0.17.4
+	k8s.io/apimachinery v0.17.4
+	k8s.io/client-go v0.17.4
+	knative.dev/client v0.14.0
```
Comment (rhuss) [inline plugins/admin/go.mod:13]: why do you need a dependency on `knative.dev/client` ?

Comment (zhanggbj) [reply plugins/admin/go.mod:13]: @rhuss
I noticed there's a method in knative.dev/client to create flags like --scale-to-zero and --no-scale-to-zero, so reused it.

```
flags.AddBothBoolFlagsUnhidden
```

**Comment by zhanggbj** [comment]
Hi @rhuss @maximilien,

All comments addressed, would you please help to review again, thanks!

File: plugins/admin/go.mod (hunk 4 above)
Comment (rhuss) [reply plugins/admin/go.mod:13]: I think this is ok for now but for the long run we should extract the parts that can be reused by plugins within the client into at least a dedicated package. Because the client coders *will not be aware* that this method is used by anybody outside the client itself, so its very likely that this can be changed by a refactoring or by accident.

**Comment by zhanggbj** [comment]
will wait for #14 to add more UT and e2e test

File: plugins/admin/pkg/command/autoscaling/update.go [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package autoscaling
+
+import (
+	"errors"
+	"fmt"
```
Comment (mattmoor) [inline plugins/admin/pkg/command/autoscaling/update.go:19]: Format Go code:
```suggestion
	"fmt"

```

File: plugins/admin/pkg/command/autoscaling/update_test.go [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+		_, err := testutil.ExecuteCommand(cmd, "--scale-to-zero")
+		assert.NilError(t, err)
+
+		updated, err := client.CoreV1().ConfigMaps(knativeServing).Get(configAutoscaler, metav1.GetOptions{})
+		assert.NilError(t, err)
+		assert.Check(t, equality.Semantic.DeepEqual(updated, cm), "configmap should not be changed")
+
+	})
+}
+
```
Comment (mattmoor) [inline plugins/admin/pkg/command/autoscaling/update_test.go:134]: Format Go code:
```suggestion

```

**Comment by zhanggbj** [comment]
As @rhuss have added lgtm and I filled up UT later, so I'll merge it.
/lgtm

**Comment by zhanggbj** [comment]
Hi @rhuss @maximilien ,

Would you please help to take a review, the last update is to fill up UT, thank you!
